cxmon/src/mon_cmd.cpp

```diff
@@ -303,6 +307,256 @@ void binary_dump(void)
+		return false;
+	}
+
+	if (mon_token != T_END) {
+		mon_error("Too many arguments");
+		return false;
+	}
+
+	if (*index_ptr > break_point_set.size()) {
+		mon_error("Illegal Index Number!");
+		return false;
+	}
+
+	return true;
+}
+
+
+/*
+ * Remove Break Point
+ */
+void break_point_remove(void)
+{
+	uintptr index;
+
+	if (!validate_index(&index, active_break_points))
+		return;
+
+	if (0 == index) {
+		active_break_points.clear();
+		printf("Removed all break points!\n");
+		return;
+	}
+
+	BREAK_POINT_SET::iterator it = active_break_points.begin();
+	std::advance(it, index - 1);
+	// Remove break point
+	printf("Removed break point %4x at address %08lx\n", index, *it);
+	active_break_points.erase(it);
+}
+
+
+/*
+ * Disable Break Point
+ */
+void break_point_disable(void)
+{
+	uintptr index;
+
+	if (!validate_index(&index, active_break_points))
+		return;
+
+	if (0 == index) {
+		for (BREAK_POINT_SET::iterator it = active_break_points.begin(); it != active_break_points.end(); it++)
+			disabled_break_points.insert(*it);
+		active_break_points.clear();
+		printf("Disabled all break points!\n");
+		return;
+	}
+
+	BREAK_POINT_SET::iterator it = active_break_points.begin();
+	std::advance(it, index - 1);
+	// Add to disable break points
+	printf("Disabled break point %4x at address %08lx\n", index, *it);
+	disabled_break_points.insert(*it);
+	// Remove break point
+	active_break_points.erase(it);
+}
+
+
+/*
+ * Enable Break Point
+ */
+void break_point_enable(void)
+{
+	uintptr index;
+
+	if (!validate_index(&index, disabled_break_points))
+		return;
+
+	if (0 == index) {
+		for (BREAK_POINT_SET::iterator it = disabled_break_points.begin(); it != disabled_break_points.end(); it++)
+			active_break_points.insert(*it);
+		disabled_break_points.clear();
+		printf("Enabled all break points!\n");
+		return;
+	}
+
+	BREAK_POINT_SET::iterator it = disabled_break_points.begin();
+	std::advance(it, index - 1);
+	// Add to active break points
+	printf("Disabled break point %4x at address %08lx\n", index, *it);
+	active_break_points.insert(*it);
+	// Remove break point
+	disabled_break_points.erase(it);
+}
+
+
+/*
+ * List all Active Break Points
+ */
+void break_point_info(void)
+{
+	if (mon_token != T_END) {
+		mon_error("Too many arguments");
+		return;
+	}
+
+	BREAK_POINT_SET::iterator it;
+
+	if (!active_break_points.empty()) {
+		int pos = 1;
+		printf(STR_ACTIVE_BREAK_POINTS);
+		for (it = active_break_points.begin(); it != active_break_points.end(); it++)
+			printf("\tBreak point %4x at address %08lx\n", pos++, *it);
+	}
+
+	if (!disabled_break_points.empty()) {
+		putchar('\n');
+		printf(STR_DISABLED_BREAK_POINTS);
+		int pos = 1;
+		for (it = disabled_break_points.begin(); it != disabled_break_points.end(); it++)
+			printf("\tBreak point %4x at address %08lx\n", pos++, *it);
+	}
+}
+
+
+/*
+ * Save all Active Break Points to a file
+ */
+void break_point_save(void)
+{
+	FILE *file;
```

Comment:
Nit: Suggest moving this right above line 488 so it's closer to where it's used.

Reply:
Done!